runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -174,6 +176,34 @@ impl<'ctx, C: Context, Cfg: Config> EVMBackend for Backend<'ctx, C, Cfg> {
     }
 }
 
+pub(crate) trait EVMBackendExt {
+    fn random_bytes(&self, num_words: u64, pers: &[u8]) -> Vec<u8>;
+}
+
+impl<T: EVMBackendExt> EVMBackendExt for &T {
+    fn random_bytes(&self, num_words: u64, pers: &[u8]) -> Vec<u8> {
+        (*self).random_bytes(num_words, pers)
+    }
+}
+
+impl<'ctx, C: Context, Cfg: Config> EVMBackendExt for Backend<'ctx, C, Cfg> {
+    fn random_bytes(&self, num_bytes: u64, pers: &[u8]) -> Vec<u8> {
+        // Refuse to generate more than 1 KiB in one go.
+        // EVM memory gas is checked only before and after calls, so we won't
+        // see the quadratic memory cost until after this call uses its time.
+        let num_bytes = num_bytes.min(RNG_MAX_BYTES) as usize;
+        let mut ctx = self.ctx.borrow_mut();
+        ctx.rng(pers)
+            .ok()
+            .and_then(|mut rng| {
+                let mut rand_bytes = vec![0u8; num_bytes];
+                rand_core::RngCore::try_fill_bytes(&mut rng, &mut rand_bytes).ok()?;
+                Some(rand_bytes)
+            })
+            .unwrap_or_default()
```

Comment:
So this will still ignore key manager errors? I would rather we either propagate the error (e.g. so the dispatcher sees the `Abort` error -- this will probably require storing it somewhere where it can be retrieved by the interpreter invocation and triggering an EVM abort) or if this is really hard to do, panic. Both will trigger an explicit failure instead of a silent discrepancy which is slashable.

Reply:
There's no `Result`-returning function on the path from here to the runtime. Precompiles don't produce errors, I guess. I wasn't aware that panicking did anything special. I'll do that instead.

Reply:
> There's no Result-returning function on the path from here to the runtime

Right, that's why I mentioned that it would probably require storing an error somewhere and triggering an EVM abort (if possible) then the caller of the interpreter can check if an error is stored and propagate that. Not the nicest, but aside from panicking it's probably the only option.

> It's the ctx.rng() one that does.

Yes, so the `ctx.rng()` is the one that can produce an error in case there is a problem with key manager communication. Deterministic errors are fine to be `.unwrap_or_default()`, it's just the non-deterministic ones that are a problem (e.g. key manager communication failing for some reason).